### DISCUSSION TITLE
Self-Hosted Docker. Resources utilization

### DISCUSSION THREAD

**Discussion by MaxGrekhov** [supabase/supabase#26159] [Questions]
I saw news about GA and a lot of buzz around Supabase lately so I decided to look at it. I work for an enterprise, and Supabase could be a way for small\mvp projects for new clients. I followed the "self-hosted docker" instructions and got the base setup. I tested performance, and it's 3-5x slower than manual implementation, which is ok for such a tool. However, I'm confused by resource usage. Docker compose stats under no load after a couple of API tests:
```
CONTAINER ID   NAME                             CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
019bc1e29c3f   supabase-storage                 0.43%     39.54MiB / 30.66GiB   0.13%     480kB / 226kB     90.1kB / 0B       11
95cf403a85a8   supabase-studio                  0.00%     71.65MiB / 30.66GiB   0.23%     4.18MB / 2.79MB   598kB / 0B        11
b7bdb79c5ce9   supabase-auth                    0.01%     8.859MiB / 30.66GiB   0.03%     287kB / 10.7kB    24.6kB / 0B       15
f5848beefe1c   supabase-meta                    0.44%     59.2MiB / 30.66GiB    0.19%     2.07MB / 2.04MB   496kB / 0B        11
eccc9521a5dd   supabase-kong                    0.08%     2.544GiB / 30.66GiB   8.30%     3.79GB / 4.23GB   586kB / 106kB     17
d8d88c4a0e09   realtime-dev.supabase-realtime   0.73%     211.2MiB / 30.66GiB   0.67%     15.8MB / 21.8MB   1.89MB / 3.55MB   57
d2dbb7a5ed10   supabase-edge-functions          0.00%     21.45MiB / 30.66GiB   0.07%     528kB / 14kB      0B / 905kB        35
3e8447ab33dc   supabase-rest                    0.10%     115.7MiB / 30.66GiB   0.37%     5.56GB / 4.76GB   0B / 0B           98
f7d3b65cc1a7   supabase-analytics               6.62%     524.7MiB / 30.66GiB   1.67%     1.02GB / 1.31GB   774kB / 3.53MB    56
633620901f5b   supabase-db                      0.40%     349.5MiB / 30.66GiB   1.11%     2.92GB / 5.16GB   7.99MB / 15GB     33
b589c390aea4   supabase-vector                  0.16%     39.86MiB / 30.66GiB   0.13%     1.76MB / 630MB    20.5kB / 0B       17
ecb49ad23518   supabase-imgproxy                0.00%     17.93MiB / 30.66GiB   0.06%     282kB / 0B        4.36MB / 0B       22
```
RAM: 
~3.1GB API gateway (kong) + analytics
~0.9GB everything other including CORE functionality
Total: ~4GB

Why do the requests router with a couple of plugins and analytics module consume so much RAM?
I googled Kong memory consumption and faced so many "leaking" issues. From a personal perspective, I would replace it instantly.
I didn't investigate analytics, but it looks quite "expensive" on RAM for analytics, although not as Kong.
Am I missing something?

**Comment by tsensei** [comment]
@MaxGrekhov Any findings on this ?

**Reply by MaxGrekhov** [reply]
I didn't dig into that. I saw an extensive routing configuration, and because I don't use Suparbase and only investigate, I'm not ready to invest in my routing configuration.
Looks like it's not ready to work out of the box.

**Comment by SOG-web** [comment]
please any solution to this

**Comment by SOG-web** [comment]
Please any update on this
